### PR TITLE
Event forms, validation, and delete dialogs

### PR DESCRIPTION
**Pull request by mgreen-simplethread**
This adds a user-facing cash flow event creation form interface with category selection and input validation.

Category selection UI - input/expense is selected using a tabbed UI at the top. User then drills down through categories and subcategories. When a category with no subcategories is selected, the user is presented with a form.

Screenshot 2020-03-02 15 40 10

I've also added dialogs that pop when the user tries to delete an event:

Screenshot 2020-02-28 13 43 26

The event creation form conforms to CFPB's design standards for field labels and validation feedback:

Screenshot 2020-03-02 12 37 16

Here's a quick demo of the new event creation flow, as well as deleting an event and all of its future recurrences:
forms-demo

### CODE REVIEW

**Comment by mgreen-simplethread** [comment]
Currently deployed to staging https://cfgov-refresh-staging.herokuapp.com/

File: cfgov/unprocessed/apps/mmt-my-money-calendar/js/views/calendar/add/category-browser.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          <li className="category-browser__tab-item">
+            <NavLink className="category-browser__tab-link" to={`/calendar/add/expense`}>Expense</NavLink>
+          </li>
+        </ul>
+      </nav>
+
+      {category.name && <h2>{category.name}</h2>}
+
+      <ul className="category-links">
+        {Object.entries(categoryOptions).map(([key, {name, subcategories, restricted}]) => !restricted && (
```
Comment (meissadia) [inline cfgov/unprocessed/apps/mmt-my-money-calendar/js/views/calendar/add/category-browser.js]: `Object.entries` does not work on IE.  I didn't see if there was a polyfill for this in the project but this is something to consider in the long run.

File: cfgov/unprocessed/apps/mmt-my-money-calendar/js/views/calendar/add/form.js [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    categoryPath = event.category;
+    category = Categories.get(categoryPath);
+    pathSegments = categoryPath.split('.');
+    eventType = pathSegments[0];
+    logger.log('Editing existing event: %O', event);
+  }
+
+  const recurrenceOptions = useMemo(() => {
+    const rules = category.recurrenceTypes ? pluck(recurrenceRules, category.recurrenceTypes) : recurrenceRules;
+    return Object.entries(rules).map(([value, { label }]) => ({ label, value }));
```
Comment (meissadia) [inline cfgov/unprocessed/apps/mmt-my-money-calendar/js/views/calendar/add/form.js]: Unsupported on IE.

File: cfgov/unprocessed/apps/mmt-my-money-calendar/js/views/calendar/add/category-browser.js (hunk 1 above)
Comment (mgreen-simplethread) [reply cfgov/unprocessed/apps/mmt-my-money-calendar/js/views/calendar/add/category-browser.js]: @meissadia I've got babel-preset-env targeting down to IE 11 during build, but let me make sure that function is actually being polyfilled

**Comment by mgreen-simplethread** [comment]
@meissadia I'm deciding now whether I should PR the first batch of work separately, or just do all of this stuff together. On one hand, it's a lot to review, but on the other, this work gives them a working calendar UI plus the ability to add events. I'm leaning toward just doing both as a single PR to CFPB and taking a more incremental approach afterwards.

File: cfgov/unprocessed/apps/mmt-my-money-calendar/js/views/calendar/add/category-browser.js (hunk 1 above)
Comment (mgreen-simplethread) [reply cfgov/unprocessed/apps/mmt-my-money-calendar/js/views/calendar/add/category-browser.js]: @meissadia It does get polyfilled during transpilation

**Comment by meissadia** [comment]
@mgreen-simplethread The goal for now is to get into the PR process.  I think it's okay as a single PR.  If they requests more incremental PRs, that'll be good feedback to have as well.

**Pull request merged**